src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionCache.java

```diff
@@ -58,36 +63,43 @@ public void ensureInputsPresent(MerkleTree merkleTree, Map<Digest, Message> addi
         Iterables.concat(merkleTree.getAllDigests(), additionalInputs.keySet());
     ImmutableSet<Digest> missingDigests =
         getFromFuture(cacheProtocol.findMissingDigests(allDigests));
-    List<ListenableFuture<?>> uploadFutures = new ArrayList<>();
+
+    List<ListenableFuture<Void>> uploadFutures = new ArrayList<>();
     for (Digest missingDigest : missingDigests) {
-      Directory node = merkleTree.getDirectoryByDigest(missingDigest);
-      if (node != null) {
-        uploadFutures.add(cacheProtocol.uploadBlob(missingDigest, node.toByteString()));
+      if (!missingDigests.contains(missingDigest)) {
```

Comment:
IIUC, `missingDigest` is an element of `missingDigests`, so this statement is always `false`. My guess is your following PRs will share `missingDigests` between actions but can we remove this check for this PR since I am not sure whether it will trigger internal checks and prevent us from submitting.

Reply:
Fixed! I'm not sure how that happened.